src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java

```diff
@@ -135,4 +178,65 @@ private static void remove(Block b) {
             hologram.remove();
         }
     }
+
+    private void createSelector(Type type, @Nonnull Block projector, @Nonnull ChestMenu menu) {
+
+        double currentOffset = DoubleHandler.fixDouble(Double.parseDouble(BlockStorage.getLocationInfo(
+                projector.getLocation(), type.parameter)));
+
+        menu.addItem(type.slot, new CustomItem(type.button.getAsItemStack(), "&7" + type.coordinate + " Offset (" + type.direction + ") &e" + currentOffset, "", "&eLeft Click: &7" + type.direction + Math.abs(type.smallChange), "&eRight Click: &7" + type.direction + Math.abs(type.largeChange)));
+        menu.addMenuClickHandler(type.slot, (pl, slot, item, action) -> {
+            double storedOffset = DoubleHandler.fixDouble(Double.parseDouble(BlockStorage.getLocationInfo(projector.getLocation(), type.parameter)));
+            double offset = action.isRightClicked() ? type.largeChange : type.smallChange;
+
+            if ((type.coordinate == 'X' || type.coordinate == 'Z') && offset + storedOffset >
+                    maxDistance.getValue()) {
+                offset = maxDistance.getValue() - storedOffset;
+            }
+
+            ArmorStand hologram = getArmorStand(projector, true);
+
+            if (type.coordinate == 'X') {
+                hologram.teleport(hologram.getLocation().add(offset, 0, 0));
+            } else if (type.coordinate == 'Y') {
+                hologram.teleport(hologram.getLocation().add(0, offset, 0));
+            } else {
+                hologram.teleport(hologram.getLocation().add(0, 0, offset));
+            }
+
+            BlockStorage.addBlockInfo(projector, type.parameter, String.valueOf(storedOffset + offset));
+            openEditor(pl, projector);
+            return false;
+        });
+    }
+
+    private enum Type {
```

Comment:
Do **NOT** declare the enum inside of a class, every class/enum should have its own file.